packages/eslint-plugin/src/rules/no-invalid-this.ts

```diff
@@ -0,0 +1,159 @@
+import baseRule from 'eslint/lib/rules/no-invalid-this';
```

Comment:
it looks like you're not actually using `baseRule` inside this rule because you had to fork the logic.
In this case we should locally define the types and the schema.

We don't want to reference the parent rule's schema here because if they add a new option, we won't have handling for it until we add it to our fork.

Locally define the `Options`/`MessageIds` to match as well - we don't need the separate module type declarations.